File: java-hashMap/Principal.java
import java.util.HashMap;
import java.util.Scanner;

public class Principal {
    public static void main(String[] args) {
        String caminho = "telephony_sessions.txt";       
        Utils utils = new Utils();
        HashMap<String, Sessao> sessoes = utils.carregarSessoes(caminho);
        System.out.println("Total de sessões lidas: " + sessoes.size());
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\nMenu:");
            System.out.println("1 - Buscar por número");
            System.out.println("2 - Listar todos");
            System.out.println("0 - Sair");
            System.out.print("Escolha uma opção: ");
            
            String opcao = scanner.nextLine().trim();            
            if (opcao.equals("0")) {
                System.out.println("Encerrando...");
                break;
            } else if (opcao.equals("1")) {
                System.out.print("Digite o número do telefone: ");
                String numero = scanner.nextLine().trim();
                Sessao sessao = sessoes.get(numero);
                if (sessao != null) {
                    System.out.println("\nSessão encontrada:");
                    System.out.println(sessao.toString());
                } else {
                    System.out.println("Número não encontrado.");
                }
            } else if (opcao.equals("2")) {
                System.out.println("\nListando todas as sessões:");
                for (String numero : sessoes.keySet()) {
                    Sessao sessao = sessoes.get(numero);
                    System.out.println(sessao.toString());
                    System.out.println("----------------------------------------");
                }
            } else {
                System.out.println("Opção inválida.");
            }
        }
        scanner.close();
    }
}

File: python-hashMap/Principal.py
from Utils import Utils

if __name__ == "__main__":
    caminho = "./telephony_sessions.txt"
    utils = Utils()
    sessoes = utils.carregarSessoes(caminho)
    print(f"Total de sessões lidas: {len(sessoes)}")

    while True:
        print("\nMenu:")
        print("1 - Buscar por número")
        print("2 - Listar todos")
        print("0 - Sair")
        opcao = input("Escolha uma opção: ").strip()

        if opcao == "0":
            print("Encerrando...")
            break
        elif opcao == "1":
            numero = input("Digite o número do telefone: ").strip()
            sessao = sessoes.get(numero)
            if sessao:
                print("\nSessão encontrada:")
                print(sessao)
            else:
                print("Número não encontrado.")
        elif opcao == "2":
            print("\nListando todas as sessões:")
            for sessao in sessoes.values():
                print(sessao)
                print("----------------------------------------")
        else:
            print("Opção inválida.")